src/Runtime/Distributions/DiscreteChar.cs

```diff
@@ -580,16 +366,8 @@ public static DiscreteChar UniformOver(params char[] chars)
         /// <returns>The created distribution.</returns>
         public static DiscreteChar UniformOver(IEnumerable<char> chars)
         {
-            Argument.CheckIfNotNull(chars, "chars");
-            
-            var rangeBounds = new List<char>();
-            foreach (char c in chars)
-            {
-                rangeBounds.Add(c);
-                rangeBounds.Add(c);
-            }
-
-            return UniformInRanges(rangeBounds);
+            Argument.CheckIfNotNull(chars, nameof(chars));
+            return Create(0, chars.Select(c => new CharRange(c, c + 1, 1)));
```

Comment:
Is it a case that Create method calls SortAndCheckRanges, whose logic can be performed more efficiently if SortAndCheckRanges knows that each character range is of size 1?

Reply:
I don't think it specializing to single chars will give significant speed-up in this function.

Anyway: SortAndCheckRanges() is called only when DiscreteChar is explicitly constructed from ranges. Which almost never happens. We usually construct it either from single chars (in which case this code is skipped) when building pointmass StringDistribution from string, or DiscreteChars get constructed through SetToProduct, SetToSum & co. These functions are implemented in "trusted code" which always creates correct sorted ranges, because of that this function is also bypassed.